core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -0,0 +1,227 @@
+package ccipsolana
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	agbinary "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// ExecutePluginCodecV1 is a codec for encoding and decoding execute plugin reports.
+// Compatible with:
+// - "OffRamp 1.6.0-dev"
+type ExecutePluginCodecV1 struct {
+}
+
+func NewExecutePluginCodecV1() *ExecutePluginCodecV1 {
+	return &ExecutePluginCodecV1{}
+}
+
+func (e *ExecutePluginCodecV1) Encode(ctx context.Context, report cciptypes.ExecutePluginReport) ([]byte, error) {
+	if len(report.ChainReports) == 0 || len(report.ChainReports) > 1 {
+		return nil, fmt.Errorf("unexpected chain report length: %d", len(report.ChainReports))
+	}
+
+	var message ccip_router.Any2SolanaRampMessage
+	chainReport := report.ChainReports[0]
+	if len(chainReport.Messages) > 0 {
+		// currently only allow committing one message at a time
+		msg := chainReport.Messages[0]
+		tokenAmounts := make([]ccip_router.Any2SolanaTokenTransfer, 0, len(msg.TokenAmounts))
+		for _, tokenAmount := range msg.TokenAmounts {
+			if tokenAmount.Amount.IsEmpty() {
+				return nil, fmt.Errorf("empty amount for token: %s", tokenAmount.DestTokenAddress)
+			}
+
+			DestTokenAddress, err := solana.PublicKeyFromBase58(string(tokenAmount.DestTokenAddress))
+			if err != nil {
+				return nil, fmt.Errorf("invalid destTokenAddress address: %w", err)
+			}
+
+			tokenAmounts = append(tokenAmounts, ccip_router.Any2SolanaTokenTransfer{
+				SourcePoolAddress: tokenAmount.SourcePoolAddress,
+				DestTokenAddress:  DestTokenAddress,
+				ExtraData:         tokenAmount.ExtraData,
+				Amount:            bigIntToBytes32(tokenAmount.Amount),
+				DestGasAmount:     bytesToUint32(tokenAmount.DestExecData),
```

Comment:
I'm not fully sure what format this field will be encoded to []byte as.
We should review this specific field with a CCIP team member.
I suspect if source chain is EVM, this may be encoded using abi.